src/proto/uprotocol/uauthority.rs

```diff
@@ -65,4 +78,51 @@ impl UAuthority {
         self.remote = Some(Remote::Id(id));
         self
     }
+
+    /// Returns whether the `Remote` `Ip` meets conformance with IPv4 or IPv6 spec
+    ///
+    /// # Returns
+    /// Returns a `Result<IpConformance, ValidationError>` where the error means remote is not IP or there is no remote
+    /// and happy path tells us whether the IP conforms with spec or not
+    ///
+    /// # Errors
+    ///
+    /// Returns a `ValidationError` in the failure case, indicating no remote or remote is not IP
+    pub fn remote_ip_conforms(&self) -> Result<IpConformance, ValidationError> {
+        if let Some(_remote) = self.remote.as_ref() {
+            match &self.remote {
+                Some(Remote::Ip(ip)) => Ok(match ip.len() {
+                    REMOTE_IPV4_BYTES => IpConformance::IPv4,
+                    REMOTE_IPV6_BYTES => IpConformance::IPv6,
+                    _ => IpConformance::NonConformal,
```

Comment:
Shouldn't this result in a ValidationError? my understanding is that this means we found an *ip* type remote that has neither 4 nor 16 bytes length ...

Reply:
I had outsourced the decision on whether non-conformal is good/bad/otherwise to the caller [like is done here](https://github.com/eclipse-uprotocol/uprotocol-rust/pull/18/files/ca15f019c2235f733944b167081bab849346c15b#r1449363340) inside of `UriValidator::validate_micro_form()`.

However, I see what you mean. I can't really imagine a scenario where the `ip` not being 4 nor 16 bytes in length would be anything other than an `ValidationError`.